File: support/src/main/java/com/wuguangxin/dialog/LoadingDialog.java
package com.wuguangxin.dialog;

import android.app.Dialog;
import android.content.Context;
import android.graphics.drawable.AnimationDrawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.view.animation.AlphaAnimation;
import android.widget.ImageView;

import com.wuguangxin.R;

/**
 * 耗时操作时显示的对话框
 *
 * <p>Created by wuguangxin on 14/8/29 </p>
 */
public class LoadingDialog extends Dialog {
	private final float DEF_ALPHA = 0.0f;
	private Context context;
	private View mView;
	private ImageView mLoadingImage;
	private AnimationDrawable anim;
	private AlphaAnimation alphaAnimation;
	private WindowManager.LayoutParams lp;
//	private TextView mLoadingMessage;
//	private String message = "处理中...";

	public LoadingDialog(Context context){
		super(context, R.style.xin_loading_dialog);
		this.context = context;
		initView();
	}

//	public LoadingDialog(Context context, String msg) {
//		super(context, R.style.xin_loading_dialog);
//		this.context = context;
//		this.message = msg;
//		initView();
//	}

	private void initView() {
		mView = LayoutInflater.from(context).inflate(R.layout.xin_dialog_loading, null);
		mLoadingImage = mView.findViewById(R.id.loading_image);
		mLoadingImage.setImageResource(R.drawable.loading); //  R.drawable.xin_loading_anim 普通转圈
//		mLoadingMessage = mView.findViewById(R.id.loading_message);
//		mLoadingMessage.setText(getMessage());
		anim = (AnimationDrawable) mLoadingImage.getDrawable();
		alphaAnimation = new AlphaAnimation(0f, 1f);
		alphaAnimation.setDuration(500);
		setCancelable(false);
		setContentView(mView);
	}

//	public String getMessage(){
//		return message;
//	}

//	public void setMessage(String message){
//		this.message = message;
//		if(mLoadingMessage != null){
//			mLoadingMessage.setText(message);
//		}
//	}

	private void startAnim() {
		// 如果未转动，就转动
		if(anim != null && !anim.isRunning()){
			anim.start();
		}
	}

	private void stopAnim() {
		// 如果正在播放 停止
		if (anim != null && anim.isRunning()) {
			anim.stop();
			mView.clearAnimation();
		}
	}

	@Override
	public void show() {
//		mView.startAnimation(alphaAnimation);
		setAlpha(DEF_ALPHA);
		startAnim();
		super.show();
	}

	public void setVisible(boolean visible) {
		if (visible) {
			show();
		} else {
			dismiss();
		}
	}

	/**
	 * 设置窗体透明度（默认0.8f）
	 * @param alpha 透明度
	 */
	public void setAlpha(float alpha) {
		Window window = getWindow();
		if (window != null) {
			lp = window.getAttributes();
			lp.alpha = alpha;
			window.setAttributes(lp);
		}
	}

	@Override
	public void hide() {
		stopAnim();
		super.hide();
	}

	@Override
	public void dismiss() {
		stopAnim();
		super.dismiss();
	}

	@Override
	public void cancel() {
		stopAnim();
		super.cancel();
	}
}

File: support/src/main/java/com/wuguangxin/dialog/LoadingDialog.kt
package com.wuguangxin.dialog

import android.app.Dialog
import android.content.Context
import com.wuguangxin.support.R
import android.graphics.drawable.AnimationDrawable
import android.view.animation.AlphaAnimation
import android.view.WindowManager
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView

/**
 * 耗时操作时显示的对话框
 *
 * Created by wuguangxin on 14/8/29
 */
class LoadingDialog(context: Context)
    : Dialog(context, R.style.xin_loading_dialog) {

    private val DEF_ALPHA: Float = 0.0f
    private var mView: View? = null
    private var mLoadingImage: ImageView? = null
    private var anim: AnimationDrawable? = null
    private var alphaAnimation: AlphaAnimation? = null
    private var lp: WindowManager.LayoutParams? = null

    init {
        initView()
    }

    private fun initView() {
        LayoutInflater.from(context).inflate(R.layout.xin_dialog_loading, null)?.let {
            mLoadingImage = it.findViewById(R.id.loading_image)
            mLoadingImage?.setImageResource(R.drawable.loading) //  R.drawable.xin_loading_anim 普通转圈
            anim = mLoadingImage?.drawable as AnimationDrawable
            alphaAnimation = AlphaAnimation(0f, 1f)
            alphaAnimation!!.duration = 500
            setCancelable(false)
            setContentView(it)
            mView = it
        }
    }

    private fun startAnim() {
        // 如果未转动，就转动
        if (anim?.isRunning == false) {
            anim?.start()
        }
    }

    private fun stopAnim() {
        // 如果正在播放 停止
        if (anim?.isRunning == true) {
            anim?.stop()
            mView?.clearAnimation()
        }
    }

    override fun show() {
        setAlpha(DEF_ALPHA)
        startAnim()
        super.show()
    }

    fun setVisible(visible: Boolean) {
        if (visible) {
            show()
        } else {
            dismiss()
        }
    }

    /**
     * 设置窗体透明度（默认0.8f）
     * @param alpha 透明度
     */
    fun setAlpha(alpha: Float) {
        window?.let { win ->
            win.attributes?.let { attr ->
                attr.alpha = alpha
                win.attributes = attr
                lp = attr
            }
        }
    }

    override fun hide() {
        stopAnim()
        super.hide()
    }

    override fun dismiss() {
        stopAnim()
        super.dismiss()
    }

    override fun cancel() {
        stopAnim()
        super.cancel()
    }
}